torch/_inductor/select_algorithm.py

```diff
@@ -366,6 +372,46 @@ def __init__(
         self.template_out: Optional[str] = None
         self.ops_handler: Optional[V.WrapperHandler] = None  # type: ignore[name-defined]
 
+        # The generated code is not dependent on the input nodes names, or symbolic sizes names.
+        # However, some of the variables returned by generate_and_load that are computed during the
+        # triton template expansions (code generation) are dependent on those.
+        # In order to cache the code generation and avoid redoing it for similar inputs that varies only by
+        # input names or symbol names, we do a record and replay method.
+        # During template expansions we record all function calls that change input_dependent_preserved_state
+        # and replay them on a cache hit to regenerate them.  Namely we want to be able to regenerate:
+        # 1) self.args.input_buffers
+        # 2) self.args.sizevars
+        # 3) self.prologue_supported_inputs
+
+        self.cached_replay_events: RecordedEventsType = []
+
+    def input_dependent_preserved_state(self) -> str:
+        return repr(
+            [
+                self.args.input_buffers,
```

Comment:
Can we just return the full repr for `self.args` ? Some templates use a workspace arg:

https://github.com/pytorch/pytorch/blob/5a422150c32f2c861061dc8be5f8d19fb6d80155/torch/_inductor/select_algorithm.py#L337-L340

Reply:
this will always end up caching ops what we do not always need and effect perf significantly.

Reply:
added  self.args.workspace_args